modules/govuk/manifests/apps/asset_manager.pp

```diff
@@ -88,7 +88,40 @@
       location ~ /raw/(.*) {
         internal;
         alias /var/apps/asset-manager/uploads/assets/$1;
-      }',
+      }
+
+      location = /cloud-storage-proxy/ {
+        # Prevent requests to this location from outside nginx
+        internal;
+
+        # Extract download host and URL from upstream HTTP headers
+        set $cloud_storage_host $upstream_http_x_cloud_storage_host;
+        set $download_url $upstream_http_x_cloud_storage_url;
```

Comment:
Sending this as part of the `X-Accel-Redirect` header might be more obvious for anyone needing to debug this later. You could then just `rewrite` out the initial `/cloud-storage-proxy`. I'm not sold on one method or the other, but it'd be good to note the reasons for this implementation style.